controllers/placementrule/manifestwork.go

```diff
@@ -123,23 +123,38 @@ func createManifestwork(c client.Client, work *workv1.ManifestWork) error {
 	}
 
 	manifests := work.Spec.Workload.Manifests
-	updated := false
+	needUpdate := false
 	if len(found.Spec.Workload.Manifests) == len(manifests) {
 		for i, m := range found.Spec.Workload.Manifests {
 			if !util.CompareObject(m.RawExtension, manifests[i].RawExtension) {
-				updated = true
+				needUpdate = true
 				break
 			}
 		}
 	} else {
-		updated = true
+		needUpdate = true
 	}
 
-	if updated {
+	if needUpdate {
 		log.Info("Updating manifestwork", namespace, namespace, "name", name)
 		work.ObjectMeta.ResourceVersion = found.ObjectMeta.ResourceVersion
```

Comment:
Can you guide me why we must set the `ResourceVersion`? Isn't it updated automatically?

Reply:
@clyang82 I think we do not need to set ResourceVersion before the update. @marcolan018 told me if we do not set this value, it will update failure. right?